### PR TITLE
Throw Exception on empty query string.  Fixes #101

### LINKED ISSUE #101: Hanging queries and fatal errors when null queries provided
Through a real rookie mistake and lack of error reporting I've spotted an issue where when sending through a null string as a query, the query will either hang with the bolt driver or throw a fatal exception with http.  I've tracked down the error to  `GraphAware\Neo4j\Client\Connection::run()`.  If `$statement` is null, I'd suggest throwing an exception.

Hopefully this will save somebody couple of the hours I spent debugging.

### PR DESCRIPTION
**Pull request by adam-cowley**

### CODE REVIEW

File: src/Connection/Connection.php [hunk 1]
```diff
@@ -107,6 +107,9 @@ public function createPipeline($query = null, $parameters = [], $tag = null)
     public function run($statement, $parameters = null, $tag = null)
     {
         $this->checkSession();
+        if (empty($statement)) {
+            throw new \InvalidArgumentException('You must provide a valid $statement parameter');
```
Comment (Nyholm) [inline src/Connection/Connection.php]: The exception message could be more clear. What do you think about changing it to

> You cannot run an empty statement.

Comment (adam-cowley) [inline src/Connection/Connection.php]: Agreed, that's much clearer

**Comment by ikwattro** [comment]
Thanks @adam-cowley , got your CLA and happy to merge after my minor comment.

**Comment by ikwattro** [comment]
Thanks @adam-cowley

**Pull request merged**